include/labwc.h

```diff
@@ -115,6 +115,14 @@ struct seat {
 		double x, y;
 	} smooth_scroll_offset;
 
+	/*
+	 * The surface whose keyboard focus is temporarily cleared with
+	 * seat_begin_server_action() and restored with
+	 * seat_finish_server_action().
+	 */
+	struct wlr_surface *last_focused_surface;
+	struct wl_listener last_focused_surface_destroy;
+
```

Comment:
Note: Perhaps this can be substituted by `server->active_view` and `seat->focused_layer`, but an advantage of this approach is that it handles all types of surfaces including unmanaged-xwayland surfaces.

Reply:
IIRC `view->active_view` isn't always in sync either (e.g. when override redirect surfaces are focused). I might be getting my wires crossed on that one though.

Reply:
Just agreeing with both - I like the last_focused_surface approach as @tokyo4j has it. The active view vs. focused surface duality is annoying but I think we have to deal with it, because of the non-view surface types.